### ISSUE TITLE
NullRefrence when doing LavaLinkConnection.Disconnect

### ISSUE THREAD

**Issue by Kriste00** [DSharpPlus/DSharpPlus#320]
# Summary
A null refrence exception can occur on `LavaLinkConnection.Disconnect`

# Details
I dont know how to recreate it, but i know the sate in which the exception happens.

Happens when `LavalinkGuildConnection.Channel` is null and `LavalinkGuildConnection.IsConnected` is true.

*Should'nt `IsConnected` be false if `Channel is null`*

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at DSharpPlus.Lavalink.LavalinkGuildConnection.get_GuildId()
   at DSharpPlus.Lavalink.LavalinkGuildConnection.get_GuildIdString()
   at DSharpPlus.Lavalink.LavalinkGuildConnection.Disconnect()
```

Here channel is null, and causes the exceprion
https://github.com/DSharpPlus/DSharpPlus/blob/5bb49222193991b5f05e366ada929670df4da3e5/DSharpPlus.Lavalink/LavalinkGuildConnection.cs#L73-L74

Is'nt this exception supposed to be thrown, since the connection is not valid if it has no Channel.
`throw new InvalidOperationException("This connection is not valid.");`

**Comment by Emzi0767** [comment]
If you can figure out the circumstances for the channel being null, I can investigate.

**Comment by Kriste00** [comment]
The only circumstances are that the `LavalinkGuildConnection` is stored in a variable and not obtained by `LavalinkNodeConnection.GetConnection`.

**Comment by Kriste00** [comment]
I switched to `LavalinkNodeConnection.GetConnection`, and now i have not seen this exception occur anymore. So i think the issue is resolved now.

**Comment by Emzi0767** [comment]
Fixed by 883e2e8525ce8ea14bcd413f8283d551f1eb1729.
